docs/cookbook/hijriDaysAdjustments.mjs

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable no-console */
+import * as Temporal from '../../polyfill/lib/temporal.mjs';
+
+/**
+ * AdjustableHijriPlainDate: A class for customizing Hijri date display.
+ *
+ * This class allows for adjustment of Hijri dates by shifting days forward or backward.
+ * It's designed to:
+ * 1. Help users synchronize displayed Hijri dates with their local Hijri calendar.
+ * 2. Facilitate the use of alternative Hijri calendar epochs.
+ *
+ * Key points:
+ * - It adjusts the display of Hijri dates without modifying the underlying calendar system.
+ * - The adjustment is a simple day shift and does not alter month lengths or other calendar rules.
+ * - This approach is useful for visual alignment but does not implement comprehensive
+ *   calendar customization or detailed Hijri calendar variants.
+ *
+ * Note: For more complex Hijri calendar adjustments or variants, a more sophisticated
+ * implementation would be necessary.
+ */
+
+class AdjustableHijriTemporal {
+  #temporal;
+  #daysToShift;
+
+  constructor(temporal, calendar = 'islamic-umalqura', daysToShift = 0) {
+    this.#temporal = temporal.withCalendar(calendar);
+    this.#daysToShift = daysToShift;
+  }
+
+  #getAdjustedDate() {
+    return this.#temporal.add({ days: this.#daysToShift });
+  }
```

Comment:
You could cache this in the constructor, e.g. `this.#adjustedTemporal = this.#temporal.add({ days: daysToShift })`. That way, you could avoid performing the same addition on every method call.

Reply:
Thank you for catching this. It should indeed be cached